server/tso/allocator_manager.go

```diff
@@ -387,3 +406,116 @@ func (am *AllocatorManager) GetAllocators(filters ...AllocatorGroupFilter) []All
 	}
 	return allocators
 }
+
+// checkPriority is used to check the election priority of a Local TSO Allocator.
+// In the normal case, if we want to elect a Local TSO Allocator for a certain DC,
+// such as dc-1, we need to make sure the follow priority rules:
+// 1. The PD server with dc-location="dc-1" needs to be elected as the allocator
+// leader with the highest priority.
+// 2. If all PD servers with dc-location="dc-1" are done, then the other PD servers
+// of DC could be elected.
+func (am *AllocatorManager) checkPriority() {
+	serverID := am.member.ID()
+	myServerDCLocation, err := am.getServerDCLocation(serverID)
+	if err != nil {
+		log.Error("skip checking allocator priority, failed to get server's dc-location",
+			zap.Uint64("server-id", serverID),
+			errs.ZapError(err))
+		return
+	}
+	// Check all Local TSO Allocator followers to see if their priorities is higher than the leaders
+	// Filter out allocators with leadership and initialized
+	allocatorGroups := am.getAllocatorGroups(FilterDCLocation(config.GlobalDCLocation), FilterAvailableLeadership())
+	for _, allocatorGroup := range allocatorGroups {
+		localTSOAllocator, _ := allocatorGroup.allocator.(*LocalTSOAllocator)
+		leaderServerID := localTSOAllocator.GetAllocatorLeader().GetMemberId()
+		// No leader, maybe the leader is not been watched yet
+		if leaderServerID == 0 {
+			continue
+		}
+		leaderServerDCLocation, err := am.getServerDCLocation(leaderServerID)
+		if err != nil {
+			log.Error("failed to get local tso allocator leader's dc-location",
+				zap.Uint64("server-id", serverID),
+				errs.ZapError(err))
+			continue
+		}
+		// For example, an allocator leader for dc-1 is elected by a server of dc-2, then the server of dc-1 will
+		// find this allocator's dc-location isn't the same with server of dc-2 but is same with itself.
+		if allocatorGroup.dcLocation != leaderServerDCLocation && allocatorGroup.dcLocation == myServerDCLocation {
+			log.Info("try to move the local tso allocator",
+				zap.Uint64("old-leader-id", leaderServerID),
+				zap.String("old-dc-location", leaderServerDCLocation),
+				zap.Uint64("next-leader-id", serverID),
+				zap.String("next-dc-location", myServerDCLocation))
+			nextLeaderKey := path.Join(am.rootPath, allocatorGroup.dcLocation, "next-leader")
+			resp, err := kv.NewSlowLogTxn(am.member.Client()).
+				If(clientv3.Compare(clientv3.CreateRevision(nextLeaderKey), "=", 0)).
+				Then(clientv3.OpPut(nextLeaderKey, fmt.Sprint(serverID))).
```

Comment:
Need to introduce the lease when we write `next-leader` into etcd, otherwise if the next leader is done, we will never have a new Local TSo Allocator anymore. This change is a low priority, I will update it later.

Reply:
Better to create an issue.

Reply:
Fixed.